src/main/java/core/basesyntax/generator/ReportGeneratorImpl.java

```
@@ -0,0 +1,22 @@
+package core.basesyntax.generator;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public void generateReport(Map<String, Integer> storage, String filePath) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+            writer.write("fruit,quantity");
+            writer.newLine();
+            for (Map.Entry<String, Integer> entry : storage.entrySet()) {
+                writer.write(entry.getKey() + "," + entry.getValue());
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

This violates checklist item #6: "Don't ignore exceptions in try-catch blocks. Let's rethrow a RuntimeException with an **informative** message and exception object."

While you are correctly re-throwing the exception, you are missing the informative message. It's better to provide context about what failed, for example: `throw new RuntimeException("Can't write report to file: " + filePath, e);`.